sig-robotics/proposal/Definition of cloud robot CRD and Operator/Definition of cloud robot CRD and Operator.md

```diff
@@ -0,0 +1,166 @@
+|   sensors   |   struct[]    |   A list of robot sensors   |
+|   resource_status   |     sruct     |   Including informations of CPU, GPU, memory, etc.  |
+
+-  The fields contained in the **task** resource can be described as the table below:
+
+|          Field          |   Type   |                 Description                 |
+| :---------------------: | :------: | :-----------------------------------------: |
+|         orderID         |   uint   | The unique identification of the order in the system, used to identify whether multiple tasks belong to the same order |
+|         taskID          |   uint   | The orderID and taskID jointly identify the task status currently executed by the robot |
+|       allocated         |  string  | Empty if not assigned, otherwise the assigned robotID |
+|  point_status_sequence  | struct[] |     Sequence of points already received and not executed     |
+| segment_status_sequence | struct[] |    Sequence of segment already received and not executed     |
+| task_destination_ID  |  uint  | Describe the task destination point ID  |
+
+- The fields contained in the **robotSync** resource can be describe as follow:
+
+|          Field          |   Type   |             Description         |
+| :---------------------: | :------: | :-----------------------------: |
+|     registed_robots     |  uint[]  |    A list of registed robotID   |
+|     last_heartbeat      |   map[string]Time   |    The last time a heartbeat packet was received  |
+
+## Controller design
+
+The Controller is mainly used to monitor the newly added custom resources in the cluster, or the changes of existing resources, and execute corresponding logic based on these informations.
+
+For *Task Controller*
+
+1. When a new task resource is created in the cluster, Task Controller should monitor this change
+2. The Task Controller assign the task to a node that can execute the task
+3. Then the Controller should update the corresponding fields of the robot and task CR
+
+For the *Robot Controller*
+
+1. The Robot Controller monitors the changes of the robot resources
+2. According to the changes, the Controller send the corresponding command to the node
+
+For the *Registration Controller*
+
+1. The Registration Controller listen for registration messages and heartbeat from robot nodes, and monitor the status of node at the same time
+2. Creating the robot CR if it receives the registration message and maintaining the informations of robotSync CR. If the node goes offline, set the DeletionTimeStamp for the corresponding robot CR
+
+The processing logic of enach Cotroller is shown in the figure below:
+
+<img src="./images/controllers.png" alt="controllers" style="zoom:60%;" />
+
+### Robot Registration
+
+<img src="./images/registration.png" alt="registration" style="zoom:70%;" />
+
+Robot registration process:
+1. Once the robot node join in the cluster, it will send heartbeat message to the cloud
```

Comment:
what is the rational to have the `robot node` but general `node` for this? if this limits the operation on `node`, that would be less flexibility for users? since they expect that `robot node` can be running  as normal  edge `node`? so that some other non-robotics services and application can be running?